source/MaterialXGenShader/ShaderNode.h

```diff
@@ -172,6 +174,12 @@ class MX_GENSHADER_API ShaderPort : public std::enable_shared_from_this<ShaderPo
     /// Return the value set on this port.
     ValuePtr getValue() const { return _value; }
 
+    /// Set a source color space for the value on this port.
+    void setColorspace(const string& colorspace) { if (!colorspace.empty()) _colorspace = colorspace; }
```

Comment:
Is it not allowed to reset a colorspace to empty string / no colorspace?

Reply:
This shouldn't be allowed. Instead of leaving it as part of the caller logic I just embedded it here.

Reply:
Removed the empty check so avoid overassumption.